fili-core/src/main/java/com/yahoo/bard/webservice/metadata/SegmentIntervalsHashIdGenerator.java

```diff
@@ -124,9 +120,11 @@ public SegmentIntervalsHashIdGenerator(
      *
      * @return A hash of the given segments
      */
-    public long getSegmentHash(Set<SortedMap<DateTime, Map<String, SegmentInfo>>> requestedSegments) {
-        return requestedSegments.stream()
-                .mapToLong(Object::hashCode)
-                .sum();
+    public Optional<Long> getSegmentHash(Stream<SortedMap<DateTime, Map<String, SegmentInfo>>> requestedSegments) {
+        return requestedSegments
+                .distinct()
+                .map(Object::hashCode)
```

Comment:
Unfortunately, `reduce` on a `LongStream` produces an `OptionalLong`, not an `Optional<Long>`, and those things are not compatible nor play well together. And the parent interface of the ID generator is a `QuerySigningService<Long>`, which requires the signature method to return `Optional<Long>`, so making the method return `OptionalLong` is a non-starter if we wish to keep the signing service able to return a different type than long (ie. string).

On the `make things more idiomatic` front, I like the idea. It ends up looking like this:

```java
if (
        querySigningService.getSegmentSetId(druidQuery)
                .map(id -> Objects.equals(cacheEntry.getMeta(), id))
                .orElse(false)
) {
    try {
```